packages/lib-aptos/index.ts

```diff
@@ -0,0 +1,86 @@
+import { DfnsApiClient, DfnsError } from '@dfns/sdk'
+import { GetWalletResponse, GenerateSignatureResponse } from '@dfns/sdk/types/wallets'
+import { Deserializer, RawTransaction, SignedTransaction } from '@aptos-labs/ts-sdk'
+
+export const hexToBuffer = (hex: string): Buffer => {
+  return Buffer.from(stripHexPrefix(hex), 'hex')
+}
+
+const stripHexPrefix = (hex: string): string => {
+  return hex.replace(/^0x/, '')
+}
+
+export type DfnsWalletOptions = {
+  walletId: string
+  dfnsClient: DfnsApiClient
+}
+
+type WalletMetadata = GetWalletResponse
+
+const assertSigned = (res: GenerateSignatureResponse) => {
+  if (res.status === 'Failed') {
+    throw new DfnsError(-1, 'signing failed', res)
+  } else if (res.status !== 'Signed') {
+    throw new DfnsError(
+      -1,
+      'cannot complete signing synchronously because this wallet action requires policy approval',
+      res
+    )
+  }
+}
+
+export class DfnsWallet {
+  private readonly dfnsClient
+
+  private constructor(private metadata: WalletMetadata, options: DfnsWalletOptions) {
+    this.dfnsClient = options.dfnsClient
+  }
+
+  public static async init(options: DfnsWalletOptions) {
+    const { walletId, dfnsClient } = options
+    const res = await dfnsClient.wallets.getWallet({ walletId })
+
+    if (res.status !== 'Active') {
+      throw new DfnsError(-1, 'wallet not active', { walletId, status: res.status })
+    }
+
+    if (res.network !== 'Aptos' && res.network !== 'AptosTestnet') {
+      throw new DfnsError(-1, 'wallet is not bound to Aptos or AptosTestnet', { walletId, network: res.network })
+    }
+
+    return new DfnsWallet(res, options)
+  }
+
+  public get address(): string {
+    return this.metadata.address!
+  }
+
+  public get publicKey(): string {
+    return this.metadata.signingKey.publicKey
+  }
+
+  // DFNS can sign:
+  //   - RawTransaction directly for simple transaction
+  //   - SignedTransaction for transction containing additional signers or fee payer (need to provide the
+  //     right transaction authenticator)
+  public async signTransaction(transaction: SignedTransaction): Promise<SignedTransaction>
+  public async signTransaction(transaction: RawTransaction): Promise<SignedTransaction>
+  public async signTransaction(transaction: RawTransaction | SignedTransaction): Promise<SignedTransaction> {
```

Comment:
another type we can support is [`AnyRawTransaction`](https://github.com/aptos-labs/aptos-ts-sdk/blob/main/src/transactions/types.ts#L354). it can make the sdk a bit friendlier to use, especially by handling the construction of `SignedTransaction` with authenticators.

* `SimpleTransaction` to wrap the rawTransaction
* `MultiAgentTransaction`, construct the right `SignedTransaction` with `AccountAuthenticatorNoAccountAuthenticator`.